api/v1alpha1/spire_server_config_types.go

```diff
@@ -0,0 +1,187 @@
+	// +kubebuilder:validation:Pattern=^[1-9][0-9]*Gi$
+	// +kubebuilder:default:="1Gi"
+	Size string `json:"size"`
+
+	// accessMode for the volume.
+	// +kubebuilder:validation:Enum=ReadWriteOnce;ReadWriteOncePod;ReadWriteMany
+	// +kubebuilder:default:=ReadWriteOnce
+	AccessMode string `json:"accessMode"`
+
+	// storageClass to be used for the PVC.
+	// +kubebuilder:validation:optional
+	// +kubebuilder:default:=""
+	StorageClass string `json:"storageClass,omitempty"`
+
+	// hostPath to be used when type is hostPath.
+	// +kubebuilder:validation:optional
+	// +kubebuilder:default:=""
+	HostPath string `json:"hostPath,omitempty"`
+}
+
+// DataStore configures the Spire SQL datastore backend.
+type DataStore struct {
+	// databaseType specifies type of database to use.
+	// +kubebuilder:validation:Enum=sql;sqlite3;postgres;mysql;aws_postgresql;aws_mysql
+	// +kubebuilder:default:=sqlite3
+	DatabaseType string `json:"databaseType"`
+
+	// connectionString contain connection credentials required for spire server Datastore.
+	// +kubebuilder:default:=/run/spire/data/datastore.sqlite3
+	ConnectionString string `json:"connectionString"`
+
+	// options specifies extra DB options.
+	// +kubebuilder:validation:optional
+	// +kubebuilder:default:={}
+	Options []string `json:"options,omitempty"`
+
+	// MySQL TLS options.
+	// +kubebuilder:default:=""
+	RootCAPath     string `json:"rootCAPath,omitempty"`
+	ClientCertPath string `json:"clientCertPath,omitempty"`
+	ClientKeyPath  string `json:"clientKeyPath,omitempty"`
+
+	// DB pool config
+	// maxOpenConns will specify the maximum connections for the DB pool.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default:=100
+	MaxOpenConns int `json:"maxOpenConns"`
+
+	// maxIdleConns specifies the maximum idle connection to be configured.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default:=2
+	MaxIdleConns int `json:"maxIdleConns"`
+
+	// connMaxLifetime will specify maximum lifetime connections.
+	// Max time (in seconds) a connection may live.
+	// +kubebuilder:validation:Minimum=0
+	ConnMaxLifetime int `json:"connMaxLifetime"`
+
+	// disableMigration specifies the migration state
+	// If true, disables DB auto-migration.
+	// +kubebuilder:default:="false"
+	// +kubebuilder:validation:Enum:="true";"false"
+	// +kubebuilder:validation:Optional
+	DisableMigration string `json:"disableMigration"`
+}
+
+// SpireServerKeyManager will contain configs for the spire server key manager
+type SpireServerKeyManager struct {
+	// diskEnabled is a flag to enable keyManager on disk.
+	// +kubebuilder:default:="true"
+	// +kubebuilder:validation:Enum:="true";"false"
+	// +kubebuilder:validation:Optional
+	DiskEnabled string `json:"diskEnabled,omitempty"`
+
+	// memoryEnabled is a flag to enable keyManager on memory
+	// +kubebuilder:default:="false"
+	// +kubebuilder:validation:Enum:="true";"false"
+	// +kubebuilder:validation:Optional
+	MemoryEnabled string `json:"memoryEnabled,omitempty"`
+}
```

Comment:
Are `DiskEnabled` and `MemoryEnabled` fields mutually exclusive? What will happen if the user sets both to true at the same time?